doc-content/enterprise-only/openshift/customimage-smartrouter-proc.adoc

```diff
@@ -0,0 +1,146 @@
+
+.Prerequisites
+
+* A JDK and Apache Maven are installed.
+* The project for deploying {PRODUCT} is created in your {OPENSHIFT} environment
+* You know the route for the {OPENSHIFT} image registry and have the permission to push images into the registry. For instructions about configuring the registry, see https://access.redhat.com/documentation/en-us/openshift_container_platform/4.5/html/registry/index[_Registry_] in {OPENSHIFT} product documentation.
+
+.Procedure
+
+. Download the `{PRODUCT_FILE}-container-sources.zip` product deliverable file from the {PRODUCT_DOWNLOAD_LINK}[Software Downloads] page of the Red Hat Customer Portal.
+. Extract the contents of the file and change to the `/src/rhpam-7-openshift-image/quickstarts/router-ext` directory.
+. Modify the sample source of the router extension in this directory as necessary. The existing code implements simple routing based on the version of the container.
+. Build the source code with Maven:
++
+[subs="attributes,verbatim,macros,quotes"]
+----
+mvn clean package
+----
++
+The build process generates the following JAR file: `target/router-ext-0.0.1-SNAPSHOT.jar` 
++
+. Create a working directory for creating the custom image, copy the generated JAR file into the directory, and then change to the directory, for example:
++
+[subs="attributes,verbatim,macros,quotes"]
+----
+mkdir /tmp/smartrouter
+cp target/router-ext-0.0.1-SNAPSHOT.jar /tmp/smartrouter
+cd /tmp/smartrouter
+----
++
+include::customimage-download-step.adoc[]
+. Extract the `openshift-launch.sh` file from the official Smart Router image:
++
+[subs="attributes,verbatim,macros,quotes"]
+----
+podman run --rm registry.redhat.io/rhpam-7/rhpam-smartrouter-rhel8:7.9.0 \
+   cat /opt/rhpam-smartrouter/openshift-launch.sh > openshift-launch.sh
+----
++
+. Edit the `openshift-launch.sh` file. In the last line of the file, find the `exec` instruction that looks like the following text:
++
+[subs="attributes,verbatim,macros,quotes"]
+----
+exec ${JAVA_HOME}/bin/java ${SHOW_JVM_SETTINGS} ${JAVA_OPTS} ${JAVA_OPTS_APPEND} ${JAVA_PROXY_OPTIONS} "${D_ARR[@]}" -jar /opt/${JBOSS_PRODUCT}/${KIE_ROUTER_DISTRIBUTION_JAR}
+----
++
+Change the instruction to the following text:
++
+[subs="attributes,verbatim,macros,quotes"]
+----
+exec ${JAVA_HOME}/bin/java ${SHOW_JVM_SETTINGS} "${D_ARR[@]}" \
+    -cp /opt/${JBOSS_PRODUCT}/router-ext-0.0.1-SNAPSHOT.jar:/opt/${JBOSS_PRODUCT}/${KIE_ROUTER_DISTRIBUTION_JAR} \
+    org.kie.server.router.KieServerRouter
+----
++
+This change adds the extension JAR file to the Java Class Path.
++
+. Create a `Dockerfile` file that defines a custom image based on the base image. The file must add the JAR file and the new `openshift-launch.sh` file, change the current user to `root`, set the necessary permissions for the `openshift-launch.sh` file, and then revert to `USER 185`, the {EAP} user. The following example shows the content of the `Dockerfile` file:
```

Comment:
Minor rewrite suggestion:

Create a `Dockerfile` file that defines a custom image based on the base image. The file must add the JAR file and the new `openshift-launch.sh` file. Change the current user to `root`, set the necessary permissions for the `openshift-launch.sh` file, and then revert to `USER 185`, which is the {EAP} user. The following example shows the content of the `Dockerfile` file:

Reply:
The file must do all this - I'll try to make it clearer.